indexing-service/src/main/java/org/apache/druid/indexing/common/task/Task.java

```diff
@@ -101,6 +101,12 @@ public interface Task
    */
   String getGroupId();
 
+  /**
+   * The group used to associate a pending segment with a task
+   * @return task group for pending segment allocations
+   */
+  String getPendingSegmentGroup();
```

Comment:
If we end up declaring this method on the `Task` interface, it should have a default implementation that throws `UnsupportedOpEx`.

Reply:
Done